pkg/controllers/testing/factory.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2020 the original author or authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import "k8s.io/apimachinery/pkg/runtime"
+
+// Factory creates Kubernetes objects
+type Factory interface {
+	// Get returns a Kubernetes object
+	Get() runtime.Object
```

Comment:
perhaps use `apis.Object` since it's the superset of `runtime.Object` and `metav1.Object`

Reply:
I'd also suggest leaving `Get()` as the typed struct, and introduce `GetObject()` to get the interface

Reply:
> I'd also suggest leaving `Get()` as the typed struct, and introduce `GetObject()` to get the interface

When would the typed struct be needed rather than the interface?

Reply:
> perhaps use `apis.Object` since it's the superset of `runtime.Object` and `metav1.Object`

Yes, very nice.